reader.go

```diff
@@ -0,0 +1,160 @@
+package gopc
+
+import (
+	"io"
+	"path/filepath"
+	"strings"
+)
+
+type archiveFile interface {
+	Open() (io.ReadCloser, error)
+	Name() string
+}
+
+type archive interface {
+	Files() []archiveFile
+}
+
+// Reader implements a OPC file reader.
+type Reader struct {
+	Parts         []*Part
+	Relationships []*Relationship
+	Properties    CoreProperties
+	p             *pkg
+	r             archive
+}
+
+// NewReader returns a new Reader reading an OPC file to r.
+func NewReader(r io.ReaderAt, size int64) (*Reader, error) {
+	zr, err := newZipReader(r, size)
+	if err != nil {
+		return nil, err
+	}
+	return newReader(zr)
+}
+
+// newReader returns a new Reader reading an OPC file to r.
+func newReader(a archive) (*Reader, error) {
+	r := &Reader{p: newPackage(), r: a}
+	if err := r.loadPackage(); err != nil {
+		return nil, err
+	}
+	return r, nil
+}
+
+func (r *Reader) loadPackage() error {
+	files := r.r.Files()
```

Comment:
This line can be moved after the next error checking

Reply:
moved